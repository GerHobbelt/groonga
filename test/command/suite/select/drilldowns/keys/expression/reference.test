table_create Names TABLE_HASH_KEY ShortText

table_create Users TABLE_HASH_KEY ShortText
column_create Users nickname COLUMN_SCALAR Names
column_create Users account_name COLUMN_SCALAR Names

load --table Users
[
{"_key": "alice1", "nickname": "alice"},
{"_key": "alice2", "account_name": "alice"},
{"_key": "bob", "nickname": "bob"}
]

select Users \
  --drilldowns[name].keys "nickname || account_name"
