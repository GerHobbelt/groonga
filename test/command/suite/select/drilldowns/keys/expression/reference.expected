table_create Names TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create Users TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Users nickname COLUMN_SCALAR Names
[[0,0.0,0.0],true]
column_create Users account_name COLUMN_SCALAR Names
[[0,0.0,0.0],true]
load --table Users
[
{"_key": "alice1", "nickname": "alice"},
{"_key": "alice2", "account_name": "alice"},
{"_key": "bob", "nickname": "bob"}
]
[[0,0.0,0.0],3]
select Users   --drilldowns[name].keys "nickname || account_name"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        3
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "account_name",
          "Names"
        ],
        [
          "nickname",
          "Names"
        ]
      ],
      [
        1,
        "alice1",
        "",
        "alice"
      ],
      [
        2,
        "alice2",
        "alice",
        ""
      ],
      [
        3,
        "bob",
        "",
        "bob"
      ]
    ],
    {
      "name": [
        [
          2
        ],
        [
          [
            "_key",
            "ShortText"
          ],
          [
            "_nsubrecs",
            "Int32"
          ]
        ],
        [
          "alice",
          2
        ],
        [
          "bob",
          1
        ]
      ]
    }
  ]
]
